tests/test_accelerator.py

```diff
@@ -703,77 +703,57 @@ def test_save_model_with_stateful_dataloader(self, use_safetensors, tied_weights
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            for step, batch in enumerate(prepared_train_dl):
+                x, y = batch
                 outputs = prepared_model(x)
                 loss = torch.nn.functional.mse_loss(outputs, y)
                 accelerator.backward(loss)
                 prepared_optimizer.step()
                 prepared_scheduler.step()
                 prepared_optimizer.zero_grad()
-            if step == num_batches_to_skip - 1:
-                state_dict = prepared_train_dl.state_dict()
-                # When breaking out without fully going through the iterator, must call end() to unregister this iterator from gradient state.
-                # TODO: Maybe this could be done automatically?
-                prepared_train_dl.end()
-                break
-
-        assert accelerator.gradient_state.active_dataloader is None
-
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            # Save model for later use
-            accelerator.save_model(model, tmpdirname, safe_serialization=use_safetensors)
-
-            # Starting from where we left off, train this model to the end of the DataLoader
-            prepared_train_dl = skip_first_batches(prepared_train_dl, num_batches_to_skip)
-            batches_seen_with_original_dl = 0
-            for batch in prepared_train_dl:
-                x, y = batch
-                x.to(accelerator.device)
-                y.to(accelerator.device)
-                with accelerator.accumulate(prepared_model):
-                    outputs = prepared_model(x)
-                    loss = torch.nn.functional.mse_loss(outputs, y)
-                    accelerator.backward(loss)
-                    prepared_optimizer.step()
-                    prepared_scheduler.step()
-                    prepared_optimizer.zero_grad()
-                batches_seen_with_original_dl += 1
-
-            original_linear1 = prepared_model.linear1.weight.clone()
-            original_batchnorm = prepared_model.batchnorm.weight.clone()
-            original_linear2 = prepared_model.linear2.weight.clone()
-
-            # Load the model and state dict
-            load_checkpoint_in_model(model, tmpdirname)
-            stateful_train_dl, _ = create_dataloaders_for_test(num_workers=num_workers)
-            prepared_stateful_train_dl = accelerator.prepare_data_loader(stateful_train_dl)
-            prepared_stateful_train_dl.load_state_dict(state_dict)
+                if step == num_batches_to_skip - 1:
+                    # Save the state once we've gone through a few batches
+                    accelerator.save_state(f"{tmpdirname}/state", safe_serialization=use_safetensors)
+                    # When breaking out without fully going through the iterator, must call end() to unregister this iterator from gradient state.
+                    # TODO: Maybe this could be done automatically?
+                    # prepared_train_dl.end()
+                    # break
```

Comment:
Should these be uncommented? If not, the comment is confusing.

Reply:
Yeah it can be removed now since we just go through the entire DL